rdt-open-reader/src/main/java/io/ona/rdt/util/Utils.java

```diff
@@ -225,4 +228,42 @@ public static boolean isValidJSONObject(String str) {
             return false;
         }
     }
+
+    public static void verifyUserAuthorization() {
+        UserAuthorizationVerificationTask.getInstance().run();
+    }
+
+    private static class UserAuthorizationVerificationTask {
+
+        private static UserAuthorizationVerificationTask INSTANCE;
+        private AsyncTask<Void, Void, Void> asyncTask;
+
+        public static UserAuthorizationVerificationTask getInstance() {
+            if (INSTANCE == null) {
+                INSTANCE = new UserAuthorizationVerificationTask();
+            }
+            return INSTANCE;
+        }
+
+        public void run() {
+            if (asyncTask == null || asyncTask.getStatus().equals(AsyncTask.Status.FINISHED)) {
```

Comment:
Move this logic to here https://github.com/onaio/rdt-standard/pull/945/files#diff-2740cd11e8b9500bd5d80204c44f77066b49757d81bdec54e40c71f60430395aR233

Reply:
done